nexus/repositories_maven_proxy.go

```diff
@@ -0,0 +1,68 @@
+//     Copyright 2020 Aicura Nexus Client and/or its authors
+//
+//     This file is part of Aicura Nexus Client.
+//
+//     Aicura Nexus Client is free software: you can redistribute it and/or modify
+//     it under the terms of the GNU Lesser General Public License as published by
+//     the Free Software Foundation, either version 3 of the License, or
+//     (at your option) any later version.
+//
+//     Aicura Nexus Client is distributed in the hope that it will be useful,
+//     but WITHOUT ANY WARRANTY; without even the implied warranty of
+//     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//     GNU Lesser General Public License for more details.
+//
+//     You should have received a copy of the GNU Lesser General Public License
+//     along with Aicura Nexus Client.  If not, see <https://www.gnu.org/licenses/>.
+
+package nexus
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// MavenProxyRepositoryService service to handle all Maven Proxy Repositories operations
+type MavenProxyRepositoryService service
+
+// Add adds new Proxy Maven repositories to the Nexus Server
+func (m *MavenProxyRepositoryService) Add(repositories ...MavenProxyRepository) error {
+	if len(repositories) == 0 {
+		m.logger.Warnf("Called AddRepository with no repositories to add")
+		return nil
+	}
```

Comment:
Adding this check we avoid creating a background thread to process the repositories list since we don't have one. It's a resource hungry task.

Instead, we could just do the work if our list is not empty, but that would hide from callers that calling the function did nothing. I'd rather do this way and give some output to the caller that something might be wrong on their side.

Reply:
> Adding this check we avoid creating a background thread to process the repositories list since we don't have one. It's a resource hungry task.

Not sure I follow... What background thread? The one inside the for loop on line 37? We wouldn't get to that if repositories is empty, the for loop would exit before any iterations.

> Instead, we could just do the work if our list is not empty, but that would hide from callers that calling the function did nothing. I'd rather do this way and give some output to the caller that something might be wrong on their side.

Yeah, I'm ok with how it is, that's a solid reason. :+1:

Reply:
> > Adding this check we avoid creating a background thread to process the repositories list since we don't have one. It's a resource hungry task.
> 
> Not sure I follow... What background thread? The one inside the for loop on line 37? We wouldn't get to that if repositories is empty, the for loop would exit before any iterations.

This line:

```go
errs, _ := errgroup.WithContext(context.Background())
```

and then here:

```
return errs.Wait()
```

There's no reason to go through this work, even if there's no thread opened.